system/Database/Postgre/Builder.php

```diff
@@ -469,9 +469,8 @@ protected function _upsertBatch(string $table, array $keys, array $values): stri
                     return ($index->type === 'UNIQUE' || $index->type === 'PRIMARY') && $hasAllFields;
                 });
 
-                foreach (array_map(static fn ($index) => $index->fields, $allIndexes) as $index) {
-                    $constraints[] = current($index);
-                    // only one index can be used?
+                foreach ($allIndexes as $index) {
+                    $constraints = $index->fields;
```

Comment:
Would $constraints be just overwritten on iterations > 1?

Reply:
No, we break the loop after the first item.